src/site_apportionment/site_apportionment.py

```diff
@@ -0,0 +1,219 @@
+import pandas as pd
+from pandas.api.types import is_numeric_dtype
+import os
+import logging
+
+SitesApportionmentLogger = logging.getLogger(__name__)
+
+# Colunm names redefined for convenience
+ref = "reference"
+ins = "instance"
+period = "period"
+form = "formtype"
+postcode = "postcodes_harmonised"
+percent = "602"
+product = "201"
+pg_num = "pg_numeric"
+civdef = "200"
+
+# Long and short form codes
+short_code = "0006"
+long_code = "0001"
+
+
+def count_unique_codes_in_col(df: pd.DataFrame, code: str) -> pd.DataFrame:
+    """Calculates the number of unique non-empty codes in a column.
+
+    Args:
+        df (pd.DataFrame): A dataframe containing all data
+        code (str): Name of the column containing codes
```

Comment:
I am a bit unsure about the `code` variable. What does it refer to (obviously a column name, but why is it called "code"?) Maybe specify what happens if the `code` column doesn't exist in the dataframe, or if it doesn't contain string data. Maybe add some error handling for these eventualities.